### ISSUE TITLE
[TT-2301] High RPS when using Tyk RRLB results in 500 errors

### ISSUE THREAD

**Issue by sedkis** [TykTechnologies/tyk#2952]
**Branch/Environment/Version**
- Branch/Version: Master
- Environment: Tyk CE On-Prem

**Describe the bug**
When Tyk is setup to do RRLB between two targets, 500 errors begin to appear when getting into the high RPS (many tens of thousands per second).

The 500 logs in GW:
```
time="Mar 13 19:55:14" level=error msg="http: proxy error: unsupported protocol scheme """ api_id=1 api_name="Tyk Test API" mw=ReverseProxy org_id=default prefix=proxy server_name=10.0.0.101 user_id="****e83b" user_ip=10.0.0.124 user_name=
```

**Reproduction steps**

1. Add api... 
Use RRLB:
```json
proxy.enable_load_balancing = true
proxy."target_list": [
        "http://<upstream1>",
        "http://<upstream2>"
  ]
```

Load test against this:

```
$ docker run --rm -it rcmorano/docker-hey -z 5m -H "Authorization:defaultaef753830b2c410b9e80912d2ed7e83b" http://10.0.0.245:8080/auth-api/json/valid

Summary:
  Total:    300.0014 secs
  Slowest:    0.2051 secs
  Fastest:    0.0002 secs
  Average:    0.0150 secs
  Requests/sec:    65043.8986

Status code distribution:
  [200]   999837 responses
  [500]    163 responses
```

163 errors in 1 million requests.  not a significant problem, yet a problem nonetheless.
I can see based on this information, it is not clear that the problem is caused by the RRLB.  Though I confirmed by removing RRLB and adding multiple times, the 500 errors only arise when this feature is enabled.

**Comment by gernest** [comment]
@sedkis I'm looking at this, can you please share the api definition file you are using?

**Comment by gernest** [comment]
No need, I was able to reproduce it

**Comment by candux** [comment]
This bug hits us way more often when we use internal tyk-targets. Is there a chance to get this merged soon?

**Comment by andyo-tyk** [comment]
This was fixed in Tyk 4.0.14, 5.0.4, 5.1.1
